sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -5277,59 +5372,224 @@ atomic_update(AccessorTy acc, Toffset byte_offset, simd_mask<N> mask) {
-    // Auto-convert FP atomics to LSC version.
-    return atomic_update<detail::to_lsc_atomic_op<Op>(), Tx, N>(acc, offset,
-                                                                src0, mask);
+  // Auto-convert FP atomics to LSC version.
+  if constexpr (L1Hint != cache_hint::none || L2Hint != cache_hint::none ||
+                Op == atomic_op::fmin || Op == atomic_op::fmax ||
+                Op == atomic_op::fadd || Op == atomic_op::fsub ||
+                !__ESIMD_DNS::isPowerOf2(N, 32)) {
+    return detail::atomic_update_impl<
+        Op, T, N, detail::lsc_data_size::default_size, L1Hint, L2Hint>(
+        acc, byte_offset, src0, mask);
   } else if constexpr (Op == atomic_op::store) {
-    if constexpr (std::is_integral_v<Tx>) {
-      return atomic_update<atomic_op::xchg, Tx, N>(acc, offset, src0, mask);
+    if constexpr (std::is_integral_v<T>) {
+      return atomic_update<atomic_op::xchg, T, N>(acc, byte_offset, src0, mask,
+                                                  props);
     } else {
-      using Tint = detail::uint_type_t<sizeof(Tx)>;
+      using Tint = detail::uint_type_t<sizeof(T)>;
       simd<Tint, N> Res = atomic_update<atomic_op::xchg, Tint, N>(
-          acc, offset, src0.template bit_cast_view<Tint>(), mask);
-      return Res.template bit_cast_view<Tx>();
+          acc, byte_offset, src0.template bit_cast_view<Tint>(), mask, props);
+      return Res.template bit_cast_view<T>();
     }
   } else {
-    detail::check_atomic<Op, Tx, N, 1>();
-    static_assert(sizeof(Tx) == 4, "Only 32 bit data is supported");
+    detail::check_atomic<Op, T, N, 1>();
+    static_assert(sizeof(T) == 4, "Only 32 bit data is supported");
     const auto si = __ESIMD_NS::get_surface_index(acc);
-    using T = typename detail::__raw_t<Tx>;
-    return __esimd_dword_atomic1<Op, T, N>(mask.data(), si, offset.data(),
-                                           src0.data());
+    using Tx = typename detail::__raw_t<T>;
+    return __esimd_dword_atomic1<Op, Tx, N>(mask.data(), si, byte_offset.data(),
+                                            src0.data());
   }
 #endif
 }
 
+/// simd<T, N>
+/// atomic_update(AccessorT acc, simd<Toffset, N> byte_offset,
+///               simd<T, N> src0, props = {});                  // (acc-au1-2)
+///
+/// A variation of \c atomic_update API with no mask operand.
+///
+/// Atomically updates \c N memory locations represented by an accessor and
+/// a vector of offsets, and returns a vector of old values found at the
+/// memory locations before update. The update operation has 1 additional
+/// argument.
+///
+/// @tparam Op The atomic operation - can be one of the following:
+/// \c atomic_op::add, \c atomic_op::sub, \c atomic_op::min, \c atomic_op::max,
+/// \c atomic_op::xchg, \c atomic_op::bit_and, \c atomic_op::bit_or,
+/// \c atomic_op::bit_xor, \c atomic_op::minsint, \c atomic_op::maxsint,
+/// \c atomic_op::fmax, \c atomic_op::fmin, \c atomic_op::store.
```

Comment:
fadd, fsub should be in this list too.

Reply:
TBH, I'm not sure it is beneficial to extensively document each function overload like this. Any mistake is being multiplied.

Reply:
IMO, it is better to have description at pages like this, rather than just a declaration. Example: https://intel.github.io/llvm-docs/doxygen/group__sycl__esimd__memory__lsc.html#ga802c4cdd9583288401c5fb4318a6f329

In this case if the list of opcodes remains, then it needs to be accurate. Please add fadd/fsub to the list.

Reply:
Yeah, this page can definitely provide value to the users, I agree.

Reply:
sorry i misunderstood what this was about, definitely the list needs to be correct. i fixed it everywhere in my latest commit